noir-projects/noir-protocol-circuits/crates/types/src/public_keys.nr

```diff
@@ -117,9 +131,19 @@ impl Eq for PublicKeys {
 }
 
 impl PublicKeys {
+    pub fn is_empty(self) -> bool {
+        PublicKeys {
+            npk_m: NpkM { inner: Point::empty() },
+            ivpk_m: IvpkM { inner: Point::empty() },
+            ovpk_m: OvpkM { inner: Point::empty() },
+            tpk_m: TpkM { inner: Point::empty() },
+        }
+            .eq(self)
+    }
+
     pub fn hash(self) -> PublicKeysHash {
         PublicKeysHash::from_field(
-            if is_empty(self) {
+            if self.is_empty() {
```

Comment:
Maybe it's worth keeping the "empty" PublicKeys implementation. It can be used in places where the address doesn't have to be derived, so that when calling hash on it, it will go into this if statement and the poseison hash will be eliminated.

Reply:
I checked the codebase again and PublicKeys is never included in a struct that can be empty. So agree with you that it's a footgun! Then maybe remove `is_empty` and the if statement in `hash`, to avoid the extra check.